cddl_gen/cddl_gen.py

```diff
@@ -2295,25 +2325,49 @@ def render_type_file(self, header_guard):
 #endif /* {header_guard} */
 """
 
-    def render(self, h_file, c_file, type_file_in):
-        h_dir, h_name = path.split(h_file.name)
-        type_file = type_file_in or open(path.join(h_dir, f"types_{h_name}"), 'w')
+    def render_cmake_file(self, target_name, h_file, c_file, type_file, output_c_dir, output_h_dir):
+        return \
+            f"""#
+# Generated using cddl_gen version {self.version}
+# https://github.com/NordicSemiconductor/cddl-gen{'''
+# at: ''' + datetime.now().strftime('%Y-%m-%d %H:%M:%S') if self.print_time else ''}
+#
+
+add_library({target_name})
+target_sources({target_name} PRIVATE
+    {Path(output_c_dir, "cbor_decode.c")}
+    {Path(output_c_dir, "cbor_encode.c")}
+    {Path(output_c_dir, "cbor_common.c")}
+    {c_file.name}
+    )
+target_include_directories({target_name} PUBLIC
+    {Path(output_h_dir)}
+    {Path(type_file.name).absolute().parent}
+    {Path(h_file.name).absolute().parent}
+    )
+"""
+
+    def render(self, h_file, c_file, type_file, cmake_file=None, output_c_dir=None,
+               output_h_dir=None):
+        _, h_name = path.split(h_file.name)
 
         # Create and populate the generated c and h file.
         makedirs("./" + path.dirname(c_file.name), exist_ok=True)
 
         print("Writing to " + c_file.name)
         c_file.write(self.render_c_file(header_file_name=h_name))
 
-        makedirs("./" + h_dir, exist_ok=True)
-        type_file_name = path.basename(type_file.name)
-
         print("Writing to " + h_file.name)
         h_file.write(self.render_h_file(
-            type_def_file=type_file_name, header_guard=self.header_guard(h_file.name)))
+            type_def_file=type_file.name, header_guard=self.header_guard(h_file.name)))
 
-        print("Writing to " + type_file_name)
-        type_file.write(self.render_type_file(header_guard=self.header_guard(type_file_name)))
+        print("Writing to " + type_file.name)
+        type_file.write(self.render_type_file(header_guard=self.header_guard(type_file.name)))
+
+        if cmake_file:
+            print("Writing to " + cmake_file.name)
+            cmake_file.write(self.render_cmake_file(
+                Path(cmake_file.name).stem, h_file, c_file, type_file, output_c_dir, output_h_dir))
```

Comment:
Should we allow the user to set the target name explicitly? Using the stem of the cmake filename might conflict with existing cmake targets in the users' project. Alternatively we could add a note in the docs about this, so that the user understands that they should rename their cmake output file to avoid the conflict.